.github/workflows/changelog-checker.yml

```diff
@@ -0,0 +1,49 @@
+# This workflow checks that there is either a 'pr/no-changelog' label applied to a PR
+# or there is a changelog/<pr number>.txt file associated with a PR for a changelog entry
+
+name: Check Changelog
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled, unlabeled]
+    # Runs on PRs to master
+    branches:
+      - master
```

Comment:
This can now be renamed to `main` I think.

Reply:
Jim's done this on `main` already, as part of his PR to do the branch switchover.